src/clusterfuzz/_internal/metrics/logs.py

```diff
@@ -576,3 +593,106 @@ def log_fatal_and_exit(message, **extras):
     info('Waiting for %d seconds before exit.' % wait_before_exit)
     time.sleep(wait_before_exit)
   sys.exit(-1)
+
+
+class GenericLogStruct(NamedTuple):
+  pass
+
+
+class TaskLogStruct(NamedTuple):
+  task_id: str
+  task_name: str
+  stage: str
+
+
+class LogContextType(enum.Enum):
+  """Log context types"""
+  TASK = 'task'
+
+  def get_extras(self) -> NamedTuple:
+    """Get the structured log for a given context"""
+    if self == LogContextType.TASK:
+      stage = log_contexts.get_metadata().get('stage', Stage.UNKNOWN).value
+      # it should exist
+      # TODO(javanlacerda): Remove this csv task_id and propagate it
+      # properly, after cheking it works in production
+      current_task_id = os.getenv('CF_TASK_ID', '').split(",")
+      task_id = current_task_id[-1]
+      task_name = current_task_id[0]
```

Comment:
Is CF_TASK_ID always in the format 'fuzz,libFuzzer,libfuzzer_asan_gopacket,f61826c3-ca9a-4b97-9c1e-9e6f4e4f8868'?

Even if it is, I would argue to have some logic to handle fails.

```suggestion
      current_task_id = os.getenv('CF_TASK_ID', '')
      match current_task_id:
        case [task_name, *_, task_id]:
          pass
        case _:
            task_name, task_id = '', ''
```

Reply:
In the task context, it is. But I agree that worth it having a try catch for avoiding breaking anything.